### ISSUE TITLE
[Table] [A11Y] Table - components in table cell headers are not readable

### ISSUE THREAD

**Issue by TheSharpieOne** [carbon-design-system/carbon-addons-iot-react#2620]
### What package is this for?

- [x] React
- [ ] Angular

### Describe the bug

===== Accessibility checkpoint =====
Test Tool Type: IBM Equal Access Accessibility Checker
Checkpoint: IBM 2.1.1 Keyboard (https://www.ibm.com/able/requirements/requirements/#2_1_1)

=====STEPS TO REPRODUCE =====

Open file storage (https://cloud.ibm.com/cloud-storage/file) or block storage (https://cloud.ibm.com/cloud-storage/block)
Login with ID & password
Use Tab key to navigate on the Filter button, tap Enter or Space to display the filter row
Continue use Tab key to navigate on location combobox
Tap Down arrow key to display the menu options
===== Problem Summary =====
When we use Tab key to navigate on location combobox and use Down arrow key to display the menu options, we cannot see the completed option strings or completed placeholder strings.

===== EXPECTED RESULT(Accessibility guidance) =====
It's not friendly to keyboard users, they can not see the completed option strings or placeholder strings.
![image](https://user-images.githubusercontent.com/664714/127173734-e36119e1-bead-4c31-af6c-71b4e68a3103.png)

**Comment by bjornalm** [comment]
It is not clear what the expected solution is.

One solution could be to use the onFocus event to show some kind of additional tooltip when the user navigates with the keyboard, but an in my opinion better solution would be to let the listbox-menu expand beyond the defined width of the column. We can have it adjust to the actual width of the menu items with a reasonable max-width which would also dramatically improve the UX for non keyboard interaction in small tables.

One possible problem with that solution is if the column is at the end of the table and the listbox-menu expands beyond the table body. In that case the table body will get a horizontal scroll. As a solution to that we could perhaps automatically scroll into position, or flip the direction of the listbox-menu.

![image](https://user-images.githubusercontent.com/5167091/131370773-08a49a20-e1ce-4f1e-b665-6aa8cb316cf4.png)

 @dianatran18 There might be another better solution, can I have your input on this?

**Comment by bjornalm** [comment]
Agreed to the suggested design after discussion with Ben Bacon.
![image](https://user-images.githubusercontent.com/5167091/131522101-d6a1aaef-46a1-4645-a7c5-7e36931a3fda.png)

**Comment by JordanWSmith15** [comment]
> One solution could be to use the onFocus event to show some kind of additional tooltip when the user navigates with the keyboard, but an in my opinion better solution would be to let the listbox-menu expand beyond the defined width of the column. We can have it adjust to the actual width of the menu items with a reasonable max-width which would also dramatically improve the UX for non keyboard interaction in small tables.
> 
> One possible problem with that solution is if the column is at the end of the table and the listbox-menu expands beyond the table body. In that case the table body will get a horizontal scroll. As a solution to that we could perhaps automatically scroll into position, or flip the direction of the listbox-menu.

@TheSharpieOne  We will deliver a fix for the dropdown menus, but for the placeholder strings, we are going to leave as-is for now and split it out into the backlog. I'll want to see if we get more instances of this being reported, as it only happens in an uncommon scenario with fairly low impact, compared to how expensive it would be to solve correctly.

**Comment by bjornalm** [comment]
@JordanWSmith15 @ben-bacon
One option to reduce the impact of the place holder issue #2788  could be to reduce the padding of the cells containing the filter elements. From this
![image](https://user-images.githubusercontent.com/5167091/131742345-f37597d5-8cd3-444d-a48a-8ce783feca99.png)
 
to this:
![image](https://user-images.githubusercontent.com/5167091/131744728-fa1004a1-0d09-4a67-928b-1f7b94d1c056.png)
